pkg/gateway/api/validation.go

```diff
@@ -127,3 +134,82 @@ func (s *gatewayService) validateMetricsEvent(ctx context.Context, id string) (s
 	}
 	return "", nil
 }
+
+func (s *gatewayService) validateLatencyMetricsEvent(
+	ctx context.Context,
+	apiID eventproto.ApiId,
+	duration time.Duration,
+) (string, error) {
+	if apiID == eventproto.ApiId_UNKNOWN_API {
+		s.logger.Warn(
+			"Failed to validate latencyMetrics event. Unkonown API id",
+		)
+		return codeUnknownAPIID, errUnknownAPI
+	}
+	if duration <= 0 {
+		s.logger.Warn(
+			"Failed to validate latencyMetrics event. Invalid duration",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.String("duration", duration.String()),
+			)...,
+		)
+		return codeInvalidDuration, errInvalidDuration
+	}
+	return "", nil
+}
+
+func (s *gatewayService) validateSizeMetricsEvent(ctx context.Context, apiID eventproto.ApiId) (string, error) {
+	if apiID == eventproto.ApiId_UNKNOWN_API {
+		s.logger.Warn(
+			"Failed to validate latencyMetrics event. Unkonown API id",
+		)
+		return codeUnknownAPIID, errUnknownAPI
+	}
+	return "", nil
+}
+
+func (s *gatewayService) validateTimeoutErrorMetricsEvent(ctx context.Context, apiID eventproto.ApiId) (string, error) {
+	if apiID == eventproto.ApiId_UNKNOWN_API {
+		s.logger.Warn(
+			"Failed to validate latencyMetrics event. Unkonown API id",
+		)
+		return codeUnknownAPIID, errUnknownAPI
+	}
+	return "", nil
+}
+
+func (s *gatewayService) validateInternalErrorMetricsEvent(
+	ctx context.Context,
+	apiID eventproto.ApiId,
+) (string, error) {
+	if apiID == eventproto.ApiId_UNKNOWN_API {
+		s.logger.Warn(
+			"Failed to validate latencyMetrics event. Unkonown API id",
+		)
+		return codeUnknownAPIID, errUnknownAPI
+	}
+	return "", nil
+}
+
+func (s *gatewayService) validateNetworkErrorMetricsEvent(ctx context.Context, apiID eventproto.ApiId) (string, error) {
+	if apiID == eventproto.ApiId_UNKNOWN_API {
+		s.logger.Warn(
+			"Failed to validate latencyMetrics event. Unkonown API id",
+		)
+		return codeUnknownAPIID, errUnknownAPI
+	}
+	return "", nil
+}
+
+func (s *gatewayService) validateInternalSdkErrorMetricsEvent(
+	ctx context.Context,
+	apiID eventproto.ApiId,
+) (string, error) {
+	if apiID == eventproto.ApiId_UNKNOWN_API {
+		s.logger.Warn(
+			"Failed to validate latencyMetrics event. Unkonown API id",
```

Comment:
All warn messages are "Failed to validate `latencyMetrics` event. Unknown API id".

Reply:
Good catch